pkg/debugapi/postage.go

```diff
@@ -321,3 +321,63 @@ func (s *Service) estimateBatchTTL(id []byte) (int64, error) {
 
 	return ttl.Int64(), nil
 }
+
+func (s *Service) postageTopUpHandler(w http.ResponseWriter, r *http.Request) {
+	idStr := mux.Vars(r)["id"]
+	if idStr == "" || len(idStr) != 64 {
+		s.logger.Error("topup batch: invalid batchID")
+		jsonhttp.BadRequest(w, "invalid batchID")
+		return
+	}
+	id, err := hex.DecodeString(idStr)
+	if err != nil {
+		s.logger.Debugf("topup batch: invalid batchID: %v", err)
+		s.logger.Error("topup batch: invalid batchID")
+		jsonhttp.BadRequest(w, "invalid batchID")
+		return
+	}
+
+	amount, ok := big.NewInt(0).SetString(mux.Vars(r)["amount"], 10)
+	if !ok {
+		s.logger.Error("topup batch: invalid amount")
+		jsonhttp.BadRequest(w, "invalid postage amount")
+		return
+	}
+
+	ctx := r.Context()
+	if price, ok := r.Header[gasPriceHeader]; ok {
+		p, ok := big.NewInt(0).SetString(price[0], 10)
+		if !ok {
+			s.logger.Error("topup batch: bad gas price")
+			jsonhttp.BadRequest(w, errBadGasPrice)
+			return
+		}
+		ctx = sctx.SetGasPrice(ctx, p)
+	}
+
+	if !s.postageCreateSem.TryAcquire(1) {
```

Comment:
consider renaming the semaphore field to just `postageSem`. also, another suggestion would be to have a new middleware for all the postage APIs that do the on-chain ops and just acquire and `defer` the release of the  resource in the middleware instead of inlining it in the different endpoints

Reply:
This seems cleaner. Will do this as a part of the dilution PR.